Added support for fan speed controllers

dtesenair opened the pull request:
Added support for the DW4SF fan speed controller.  This is accomplished by reading the "customType" attribute on the device properties returned from the MyLeviton service.  I created separate functions to set up Lightbulb and Fan services and called the appropriate function via a switch statement. I did this in hopes of adding support for various different device types (switches, etc).  However, the only device I have to test on is the DW4SF.

I'm fairly new to JavaScript development so please excuse any bad coding.  Feel free to correct anything you see fit.

Thanks for writing this plugin!

tabrindle commented:
Hey this is awesome! Buying one of these as we speak to test this out, thanks a bunch for your contribution! Nothing jumps out at me quickly looking at the code, so great job.

dtesenair commented:
So, I just realized that the "customType" attribute returned from the API call represents the icon you have chosen in the My Leviton app and DOES NOT represent the actual device type.  As such, I rewrote the switch statement to use the device model number to determine the service type. Also added the "Switch" and "Outlet" services for the corresponding models.

tabrindle commented:
That makes sense!

I think this pattern will also help with our other outstanding issue. Thanks for your patience. I have one of these devices installed, just haven't tested yet. As soon as I verify this on my end, I will merge your PR and get a new build out.

tabrindle commented on index.js:
```diff
@@ -201,31 +247,152 @@ class LevitonDecoraSmartPlatform {
+    this.log('  -Device Model: ', device.model)
+    
+    
+    switch (device.model) {
```
This switch should take advantage of case fall through to make it more compact and easy to read. 

> This is an example of a single operation sequential case statement, where four different values perform exactly the same.
```
var Animal = 'Giraffe';
switch (Animal) {
  case 'Cow':
  case 'Giraffe':
  case 'Dog':
  case 'Pig':
    console.log('This animal will go on Noah\'s Ark.');
    break;
  case 'Dinosaur':
  default:
    console.log('This animal will not.');
}
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

dtesenair replied:
Just issued another commit with your suggestion on the case fall through.  Thanks for that pointer.

charlieschmidt commented:
As a data point, it also works perfectly on my DW4SF switch.  Thanks to you both!

tabrindle commented:
Published as v1.1.0~
